scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py

```diff
@@ -96,5 +98,85 @@ def get_pair_data_directory(context, run):
         return context.data_context._base_run_name(run) + " Pairs/"
 
 
-def get_phase_quad_workspace_name(context, run, period):
-    return get_raw_data_workspace_name(context, run, period=str(period)) + " (PhaseQuad)"
+# def get_phase_quad_workspace_name(context, run, period):
+#     return get_raw_data_workspace_name(context, run, period=str(period)) + " (PhaseQuad)"
+
+
+def calculate_base_name_and_group_for_phase_table(context, table_name):
+    run = re.search('[0-9]+', table_name).group()
+    base_name = table_name + '; ' + context.phase_context.options_dict['forward_group']
+    base_name += ', ' + context.phase_context.options_dict['backward_group']
+
+    group = context.data_context._base_run_name(run) + ' PhaseTables'
+
+    if not AnalysisDataService.doesExist(group) or type(AnalysisDataService.retrieve(group)) is not WorkspaceGroup:
+        new_group = WorkspaceGroup()
+        AnalysisDataService.addOrReplace(group, new_group)
+        AnalysisDataService.addToGroup(context.data_context._base_run_name(run), group)
+
+    return base_name, group
+
+
+def calculate_base_name_and_group_for_phasequad(context, input_workspace, phase_table):
+    base_name = input_workspace.split('_')[0] + '; PhaseQuad; ' \
+                + phase_table
+
+    run = re.search('[0-9]+', input_workspace).group()
+    group = context.data_context._base_run_name(run) + ' PhaseTables'
+
+    if not AnalysisDataService.doesExist(group) or type(AnalysisDataService.retrieve(group)) is not WorkspaceGroup:
```

Comment:
Is it worth making this a function as it seems to be used in a few places i.e. `if some_name(group):`?